contracts/actions/compound/CompPayback.sol

```diff
@@ -56,25 +65,31 @@ contract CompPayback is ActionBase, CompHelper {
     function _payback(
         address _cTokenAddr,
         uint256 _amount,
-        address _from
+        address _from,
+        address _onBehalf
     ) internal returns (uint256) {
         address tokenAddr = getUnderlyingAddr(_cTokenAddr);
 
-        uint256 maxDebt = ICToken(_cTokenAddr).borrowBalanceCurrent(address(this));
+        // default to onBehalf of proxy
+        if (_onBehalf == address(0)) {
+            _onBehalf = address(this);
+        }
+
+        uint256 maxDebt = ICToken(_cTokenAddr).borrowBalanceCurrent(_onBehalf);
         _amount = _amount > maxDebt ? maxDebt : _amount;
 
         tokenAddr.pullTokensIfNeeded(_from, _amount);
 
         // we always expect actions to deal with WETH never Eth
         if (tokenAddr != TokenUtils.WETH_ADDR) {
             tokenAddr.approveToken(_cTokenAddr, _amount);
-            require(ICToken(_cTokenAddr).repayBorrow(_amount) == NO_ERROR, ERR_COMP_PAYBACK_FAILED);
+            require(ICToken(_cTokenAddr).repayBorrowBehalf(_onBehalf ,_amount) == NO_ERROR, ERR_COMP_PAYBACK_FAILED);
```

Comment:
Are we sure behavior is exactly the same in `repayBorrow` and `repayBorrowBehalf `

Reply:
according to their docs yes https://compound.finance/docs/ctokens#repay-borrow-behalf